tests/integration/test_ask_tell_optimization.py

```diff
@@ -0,0 +1,150 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+import pickle
+from typing import Callable, List, Tuple, Union, cast
+
+import gpflow
+import numpy.testing as npt
+import pytest
+import tensorflow as tf
+import tensorflow_probability as tfp
+
+from tests.util.misc import random_seed
+from trieste.acquisition.function import LocalPenalizationAcquisitionFunction
+from trieste.acquisition.rule import AcquisitionRule, EfficientGlobalOptimization, TrustRegion
+from trieste.ask_tell_optimization import AskTellOptimizer
+from trieste.bayesian_optimizer import OptimizationResult, Record
+from trieste.data import Dataset
+from trieste.models.gpflow import GaussianProcessRegression
+from trieste.objectives import (
+    BRANIN_MINIMIZERS,
+    BRANIN_SEARCH_SPACE,
+    SCALED_BRANIN_MINIMUM,
+    scaled_branin,
+)
+from trieste.objectives.utils import mk_observer
+from trieste.observer import OBJECTIVE
+from trieste.space import Box, SearchSpace
+from trieste.types import State, TensorType
+
+
+@random_seed
+@pytest.mark.parametrize(
+    "num_steps, reload_state, acquisition_rule_fn",
```

Comment:
If the test isn't too slow then it might make sense to just parametrize `reload_state` separately, so each test case gets run once with reload_state and the other without. If it is slow, then you should consider whether we really want to run all the permutations.

Reply:
that's a good point. i guess we can remove one permutation indeed